main.tf

```diff
@@ -46,42 +46,58 @@ resource "google_compute_region_backend_service" "default" {
   protocol         = "${var.ip_protocol}"
   timeout_sec      = 10
   session_affinity = "${var.session_affinity}"
-  backend          = ["${var.backends}"]
+  dynamic "backend" {
+    iterator = group
+    for_each = var.backends
+    content {
+      group = group.value["group"]
+    }
+  }
   health_checks    = ["${element(compact(concat(google_compute_health_check.tcp.*.self_link,google_compute_health_check.http.*.self_link)), 0)}"]
 }
 
 resource "google_compute_health_check" "tcp" {
-  count = "${var.http_health_check ? 0 : 1}"
+  count = "${var.health_check["type"] == "tcp" ? 1 : 0}"
   project = "${var.project}"
   name    = "${var.name}-hc"
-
   tcp_health_check {
     port = "${var.health_port}"
   }
 }
 
 resource "google_compute_health_check" "http" {
-  count = "${var.http_health_check ? 1 : 0}"
+  count = "${var.health_check["type"] == "http" ? 1 : 0}"
   project = "${var.project}"
   name    = "${var.name}-hc"
-
+  check_interval_sec = "${var.health_check["check_interval_sec"]}" # - (Optional) How often (in seconds) to send a health check. The default value is 5 seconds.
+  healthy_threshold = "${var.health_check["healthy_threshold"]}" # - (Optional) A so-far unhealthy instance will be marked healthy after this many consecutive successes. The default value is 2.
+  timeout_sec = "${var.health_check["timeout_sec"]}" # - (Optional) How long (in seconds) to wait before claiming failure. The default value is 5 seconds. It is invalid for timeoutSec to have greater value than checkIntervalSec.
+  unhealthy_threshold = "${var.health_check["unhealthy_threshold"]}" # - (Optional) A so-far healthy instance will be marked unhealthy after this many consecutive failures. The default value is 2.
   http_health_check {
     port = "${var.health_port}"
+    # host = "${var.http_health_check.host != null ? http_health_check.port : null}" # - (Optional) The value of the host header in the HTTP health check request. If left empty (default value), the public IP on behalf of which this health check is performed will be used.
+    request_path = "${var.http_health_check["request_path"]}" #  - (Optional) The request path of the HTTP health check request. The default value is /.
+    host = "${var.http_health_check["host"]}" # - (Optional) The value of the host header in the HTTP health check request. If left empty (default value), the public IP on behalf of which this health check is performed will be used.
+    response = "${var.http_health_check["response"]}" #  - (Optional) The bytes to match against the beginning of the response data. If left empty (the default value), any response will indicate health. The response data can only be ASCII.
```

Comment:
`response` `port_name`, and `proxy_header` should be configured on both `google_compute_health_check` resources.